lib/auth/init.go

```diff
@@ -842,7 +842,9 @@ func createPresetRoles(ctx context.Context, rm PresetRoleManager) error {
 
 		role := role
 		g.Go(func() error {
-			if types.IsSystemResource(role) {
+			// Specifically skip the Okta requester role, as it will be
+			// modified by the Okta access list sync.
+			if types.IsSystemResource(role) && role.GetName() != teleport.SystemOktaRequesterRoleName {
```

Comment:
The mixing of concepts seems weird to me, as it'll go against the intuitions we gather from both system and preset roles. That said idk what the product goals are, so this is just a comment.

Do we need a new name for what "okta-requester" is now? "system-preset"?

Reply:
Yeah this feels like it will lead to a lot of confusion down the road.

Reply:
I don't disagree. I had initially gone down the road of having a "dynamic" system resource vs. a "static" system resource, but at the moment this is a singular exception that we don't know that we're going to expand on. There's also a possibility that if I introduce a new type or subtype here, it might get accidentally copied by the next preset introduced and cause unexpected behavior.

My argument here would be that, in the near term, this is the only known exception here, so this is preferable to introducing a whole new type until we've got a more identifiable pattern.